src/Nest/Aggregations/AggregationMetadata.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Nest
+{
+	internal static class AggregationMetadata
+	{
+		public static readonly string Key = "_type";
+
+		public static string Value(IAggregation aggregation)
+		{
+			var interfaces = aggregation.GetType().GetInterfaces();
+			foreach(var i in interfaces)
+			{
+				var value = GetElasticsearchName(i);
+				if (Map.ContainsKey(value))
+					return value;
+			}
+			return null;
+		}
+
+		private static Dictionary<string, Type> _map;
+		public static Dictionary<string, Type> Map
```

Comment:
This property isn't thread-safe; the map initialization could be done in a static constructor.

I think it might make sense to also return a `ReadOnlyDictionary<string, Type>`?